pkg/scheduler/context.go

```diff
@@ -590,8 +590,52 @@ func (cc *ClusterContext) updateNodes(request *si.UpdateRequest) {
 func (cc *ClusterContext) addNodes(request *si.UpdateRequest) {
 	acceptedNodes := make([]*si.AcceptedNode, 0)
 	rejectedNodes := make([]*si.RejectedNode, 0)
+	// if we have an unlimited node, reject all new nodes
+	// if we have an unlimited node between the new nodes, register only that node
```

Comment:
This needs to be moved into the loop: nodes are linked to partitions. I _could_ have an unlimited node on one partition but not on the other:
- create a new node object
- get the partition: fail if partition not found (existing behaviour)
- check for the unlimited and limited combination:
  - if new node is unlimited registered partition nodes must be 0
  - if new node is limited registered nodes = 0 or larger than 1 OK otherwise check the single node and make sure it is not unlimited
- reject / accept nodes are processed as per the normal flow from this point

Reply:
It is in the loop. I can moove itt after we check the partition, however we need some checks what did not depend on the partition, such as check if reservation is enabled, and check if theerare other nodes in teh `NewSchedulableNodes` list. I think we need this check as well, because if we check only the already registered nodes, the processing of the nodes will depend on the processing order and we might have different behaviours with the same imput list, what is not good.

I will moove the already reeegistered node check after we have the partition, and I will make it a per partition check, and I will keep the other checks at the beginning, since we don't need the partition for those checks and we can fail it earlier.